Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -23,10 +23,90 @@
 from abc import abstractmethod
 from distutils.version import LooseVersion
 from threading import Lock
+from inspect import currentframe
 
 import demistomock as demisto
 import warnings
 
+def __line__():
+    cf = currentframe()
+    return cf.f_back.f_lineno
+
+_MODULES_LINE_MAPPING = {
+    'CommonServerPython': {'pre': __line__() - 36, 'post': float('inf')},
+}
+
+def register_module_line(module_name, pre_post, line):
+    """
+        Register a module in the line mapping for traceback line correction algorithm.
+
+        :type module_name: ``str``
+        :param module_name: the name of the module. (required)
+
+        :type pre_post: ``str``
+        :param pre_post: Whether to register the line as the start or the end of the module. (required)
+
+        :type line: ``int``
+        :param line: the line number to record. (required)
+
+        :return: None
+        :rtype: ``None``
+    """
+    global _MODULES_LINE_MAPPING
+    try:
+        if pre_post not in ('pre', 'post'):
+            raise ValueError('Invalid pre_post argument. Acceptable values are: pre, post.')
+        if not isinstance(line, int):
+            raise ValueError('Invalid line argument. Expected int got {}'.format(type(line)))
+
+        _MODULES_LINE_MAPPING.setdefault(module_name, {'pre': 0, 'post': float('inf')}).update({pre_post: line})
+    except Exception as exc:
+
+        demisto.debug('failed to register module line. '
+            'module: "{}" pre_post: "{}" line: "{}".\nError: {}'.format(module_name, pre_post, line, exc))
+
+
+def _find_relevant_module(line):
+    global _MODULES_LINE_MAPPING
+
+    relevant_module = ''
+    for module, info in _MODULES_LINE_MAPPING.items():
+        if info['pre'] <= line <= info['post']:
+            if not relevant_module:
+                relevant_module = module
+            elif info['pre'] > _MODULES_LINE_MAPPING[relevant_module]['pre']:
+                relevant_module = module
+
+    return relevant_module
+
+
+def fix_traceback_line_numbers(trace_str):
+    """
+    Fixes the given traceback line numbers.
+
+    :type trace_str: ``str``
+    :param trace_str: The traceback string to edit. (required)
+
+    :return: The new formated traceback.
+    :rtype: ``str``
+    """
+    for number in re.findall('line (\d+)', trace_str):
+        line_num = int(number)
+        module = _find_relevant_module(line_num)
+        if module:
+            module_start_line = _MODULES_LINE_MAPPING.get(module, {'pre':0})['pre']
+            actual_number = line_num - module_start_line
+
+            # in case of ApiModule injections, adjust the line numbers of the code after the injection.
```

Comment:
Is there a way to handle `ApiModule`s that are at the beginning?
Maybe in a future phase?

Reply:
not sure what do you mean here?
the code shouldn't assume the location of the import (but I'm still testing it)